pkg/frontend/querymiddleware/shard_active_series_test.go

```diff
@@ -440,19 +442,19 @@ func Test_shardActiveSeriesMiddleware_mergeResponse_contextCancellation(t *testi
 		{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewReader(body))},
 	}
 
+	cancelCause := "request canceled while streaming response"
+	cancel(fmt.Errorf(cancelCause))
+
 	resp := s.mergeResponses(ctx, responses, "")
 	defer func() {
 		_, _ = io.Copy(io.Discard, resp.Body)
 		_ = resp.Body.Close()
 	}()
 
 	var buf bytes.Buffer
-	_, err = io.CopyN(&buf, resp.Body, int64(os.Getpagesize()))
+	_, err = io.Copy(&buf, resp.Body)
```

Comment:
Why are these changes needed?

Reply:
I believe it was due to a side effect caused by the fact that now the data value is preprocessed before performing the merge operation, but since this logic is going to change due to https://github.com/grafana/mimir/pull/7665#discussion_r1535163838, I'll restore the old code.